### ISSUE TITLE
Feature/Link element white version

### ISSUE THREAD

**Issue by Kushagra-ag** [razorpay/blade#1165]
Need to implement the navbar of frontend-website which consists of white colored link elements (which turn blue on hover/focus). 
<img width="867" alt="Screenshot 2023-05-11 at 10 32 30 AM" src="https://github.com/razorpay/blade/assets/31826645/8f5a87ae-1a6f-4adb-bcb2-14c058be57a3">

Currently this is not possible in blade as the default color of Link component is #80B3F9 (light blue)

A variant prop is needed with a version of Link component that supports white coloured text.

**Comment by saurabhdaware** [comment]
@saurav12 I think we had discussed about the white links few days back right? for the places where background is dark? are we planning to add them on design?

**Comment by saurav12** [comment]
No, we will not be adding any such variants for now.

**Comment by saurabhdaware** [comment]
@Kushagra-ag I think you can use custom Link from your repo for now. Some of the designs might change with new redesign so we can take a look at this again then.

Just wondering, apart from visual things, do you see any other blockers to use Link component from blade for such usecases? Anything from functionality-side or any prop that you would need to implement such menu that doesn't exist right now?

**Comment by divyanshu013** [comment]
Could you please also check with your design team if the banking dark theme link can be used? This would be supported by Blade's link. [Docs](https://blade.razorpay.com/?path=/docs/components-link--default&globals=themeTokenName:bankingTheme;colorScheme:dark)

![image](https://github.com/razorpay/blade/assets/6682655/4f607a6d-5754-4464-bf8d-3dbfcb618fb1)

**Comment by Kushagra-ag** [comment]
@saurav12 how do you propose to move ahead with migrating current version of navbar to blade. You have shared this [figma](https://razorpay.slack.com/archives/CTM086NSF/p1681210017388359?thread_ts=1681207425.563069&cid=CTM086NSF) with me but in here, you have used the heading text component for nav link elements (screenshot attached in issue description) instead of the link component.

**Comment by saurav12** [comment]
@Kushagra-ag This was done a while back. In the new version, there is a body token that is being used. I've made all those changes to Figma as well, please check.

Also, please check the Blade version. All the Chakra version has the screenshot attached as there is no point in re-creating everything for Chakra.

**Comment by Kushagra-ag** [comment]
@saurav12 saw the changes. There's some confusion I think.
The body variant can be used with the `Text` component. Here we will need to use the `Link` component which doesn't support body token.

**Comment by saurav12** [comment]
@Kushagra-ag The Link token also has 3 variants:
- `Small` (12px)
- `Medium` (14px)
- `Large` (16px)

Are you talking about some other variants?

**Comment by Kushagra-ag** [comment]
Discussed over call. Saurav will get back with some solution if possible without adding another variant.

> Just wondering, apart from visual things, do you see any other blockers to use Link component from blade for such usecases? Anything from functionality-side or any prop that you would need to implement such menu that doesn't exist right now?

@saurabhdaware I do have one question for the button component
For scenarios like this (login btn navbar) - 
<img width="107" alt="Screenshot 2023-05-11 at 5 18 28 PM" src="https://github.com/razorpay/blade/assets/31826645/53da4d7c-c6ed-4c00-b924-9eeacb797fe2">

We do not support `to` or `href` attribute. What should be the way to go in this case to redirect the user to login page?

**Comment by saurav12** [comment]
> Discussed over call. Saurav will get back with some solution if possible without adding another variant.

@Kushagra-ag You will need to create a custom component for this. This is not supported in Blade.


> For scenarios like this (login btn navbar) -

I think in the new designs I proposed, this `secondary` variant of the `Button` component has been used.
<img width="229" alt="Screenshot 2023-05-11 at 5 25 00 PM" src="https://github.com/razorpay/blade/assets/4298450/f2a0c93e-2cf2-44bc-aaad-664cc0498dde">

**Comment by Kushagra-ag** [comment]
No, it doesn't matter which variant of button is used. The button component in blade doesn't support `to` or `href` attribute. My question is how do I redirect to another page using button component from blade (Is `onClick` the only option here)

@saurabhdaware @divyanshu013

**Comment by saurabhdaware** [comment]
Created #1166 for tracking Button with href. This one can be used for tracking the white link only.

**Comment by kamleshchandnani** [comment]
@Kushagra-ag This is a custom case and pattern not generalized across the org. So if you want to migrate the existing chakra navbar you can easily create this using our tokens. This isn't just a link component because it has other things like when you hover the dropdown opens, similarly when you hover there's a dash that appears below Nav Bar Item. so semantically its not even link

**Comment by iAmServerless** [comment]
@saurav12 @kamleshchandnani How can we ensure that designs are also restricted to design guideline if design guideline do not permit changes?

**Comment by chaitanyadeorukhkar** [comment]
> @saurav12 @kamleshchandnani How can we ensure that designs are also restricted to design guideline if Blade do not permit changes?

@iAmServerless Some of these are tool limitations. Figma allows designers to detach components, the fact that we've added restrictions on code helps us to stop such diversions from being implemented. We've also formed a designer group called Blade Advocates where we're constantly driving education efforts and getting components & their restrictions reviewed.
